### DISCUSSION TITLE
Support find and replace within a cell (was: Assigning shortcut to replace within cell)

### DISCUSSION THREAD

**Discussion by PastorJordi** [microsoft/vscode-jupyter#9402] [New Ideas]
Hi there,
First of all, thanks for your support and the amazing job you are doing.

I've started using python notebooks with vscode recently (I am still tweaking shortcuts and this kind of stuff). 
I was wondering if I could do "Find and Replace" inside a **single cell**  without touching my mouse (like I can in a regular python script).

(Using the default settings)
`Ctrl` + `F` / `Ctrl` + `H` toggle the find/replace palette for the **entire** notebook, and the `enter` (or `ctrl`+`alt`+`enter`) key to replace does not work in this case.
Alternatively, pressing `F3` while editing a cell  with a word highlighted, pops a find menu **within** that cell, searching for the highlighted word. `Enter` or `F3` work fine to find next matches, however, typing `ctrl` + `h` will pop the replace for the entire notebook, nor I can find an alternative way of toggling 'replace' within a cell without using my mouse (`Enter` or `Ctrl` + `Alt` + `Enter` works in this scenario).

I've spent some time trying to figure out a way to be able to use both find/replace scopes (whole nootebook and single cell, respectively) only with keystrokes (even creating macros, using the geddski's extension) without success.

Do you have any advice to achieve it?

Thanks,

PS: somehow related: find and replace in notebook scope, starts cycling over the matches from the beginning every single time there's a replacement instead of jumping the the next match (like it does in a regular script). Do you think I should open an issue?

**Comment by amunger** [comment]
It looks like there is a [feature request in VS Code](https://github.com/microsoft/vscode/issues/141493) for this, although, not specifically for keyboard shortcuts. Find and replace for notebooks is a fairly new feature implemented by @rebornix, so it's great to have your feedback!

I'm pretty sure I remember triaging an issue for your final comment as well, but I can't seem to find it. (It would also be in VS Code)
